travel-rx/fe-travel-rx
# Med details

## What does this PR do?
- Adds `MedDetails` screen with (non functioning) delete button
- This screen is accessed by clicking on any medication from the `MedicineCabinet`
- Adjusts some of the styling, formatting, sizing to fit on a smaller screen size
- Deletes `FAQ`, we can access this from past branches for future iterations if needed
- Updates mock Meds data for passing props and appending the information, this includes adding an id, as we will need it for `delete` and `edit`
- Dismisses keyboard when `Search for Generic Name` is clicked

### Where should the reviewer start?
- Check the `MedDetails.js` file

### How should this be manually tested?
- Take a run through the app and see how the format/styling looks on your larger phone

### Card(s) closed:
Closes #36 

### Screenshots:
![Image from iOS (1)](https://user-images.githubusercontent.com/44537724/71874716-d6ee4080-30df-11ea-9a2a-61fd2da75def.png)
### Additional notes/questions:
- I am not wild about the  `MedDetails` styling, let me know your thought on this one
- I removed the `fontWeight: 'bold'` in a few places, the bold text was just too much on my smaller screen and also made the Chinese and Arabic difficult to read I think
- I added quotes around the English of the translated words, also not sure if I like it or not. Maybe we can ask Quinne to take a look at a few styling things and get her feedback??
- I got some great feedback via Slack from David about our different languages. Most of it is good but we can review what he shared when we meet.
- I did a bit of research about the `prop-type warnings` we are getting and think we may need to convert the values to strings (sounds annoying but here are a couple of resources we can look at and maybe ask around)
https://stackoverflow.com/questions/46984544/redux-form-invalid-prop-value-of-type-number-supplied-to-textinput-expec
https://aboutreact.com/warning-invalid-prop-value-of-type-number-supplied-to-textinput-expected-string/

### Helpful Resources
- Passing props through navigation https://blog.usejournal.com/react-navigation-cheatsheet-bf99f09d8060
- Adds `x` to clear input field https://stackoverflow.com/questions/45249807/clear-react-native-textinput
- Clear/hide keyboard https://stackoverflow.com/questions/29685421/hide-keyboard-in-react-native
